Release v23

JuiP opened the pull request:
- [x] Fix read_file method for resume from journal

- [x] Add screenshots of solutions all levels, could be used for adding a help button functionality later.

- [x] Other changes to release the activity

JuiP committed:
load_save is no longer required after commit 595df8f

JuiP committed:
Remove unused imports

JuiP committed:
Class variable and class function differenciate naming

JuiP committed:
Part of python3 port changes
- Fix the variable names, consistent with other activities

JuiP committed:
Fixes
- Fixed undefined name error for function calls
- Changed if statement comparisions with None

JuiP committed:
Added Screenshots of solutions

JuiP commented:
@srevinsaju @Saumya-Mishra9129 @chimosky @quozl Please test and review changes so I can go ahead with the release :)

srevinsaju commented on IQ.py:
```diff
@@ -349,17 +349,17 @@ def shuffle(self, lst):
-    def centre_blit(self, screen, img, xxx_todo_changeme, angle=0):  # rotation is clockwise
-        (cx, cy) = xxx_todo_changeme
+    def centre_blit(self, screen, img, coordinates, angle=0):  # rotation is clockwise
+        (cx, cy) = coordinates
```
redundant parenthesis may be removed (optional)

JuiP replied:
Thanks for noticing! :)
Done in ad31fc5

srevinsaju commented on IQ.py:
```diff
@@ -349,17 +349,17 @@ def shuffle(self, lst):
-        (r, g, b) = xxx_todo_changeme2
+    def text_blit(self, screen, s, font, coordinates, color, shadow=True):
+        (cx, cy) = coordinates
+        (r, g, b) = color
```
same

JuiP replied:
Done in ad31fc5

srevinsaju commented on IQ.py:
```diff
@@ -373,9 +373,9 @@ def text_blit(self, screen, s, font, xxx_todo_changeme1, xxx_todo_changeme2, sha
-        (r, g, b) = xxx_todo_changeme4
+    def text_blit1(self, screen, s, font, coordinates, color, shadow=True):
+        (x, y) = coordinates
+        (r, g, b) = color
```
same

JuiP replied:
Done in ad31fc5

srevinsaju commented on IQ.py:
```diff
@@ -393,8 +393,8 @@ def text_blit1(self, screen, s, font, xxx_todo_changeme3, xxx_todo_changeme4, sh
-    def message(self, screen, font, m, xxx_todo_changeme5=(0, 0), d=20):
-        (cx, cy) = xxx_todo_changeme5
+    def message_func(self, screen, font, m, coordinates=(0, 0), d=20):
+        (cx, cy) = coordinates
```
this

JuiP replied:
Done in ad31fc5

srevinsaju reviewed (commented):
I found minor changes; Its entirely optional. I will do activity testing tomorrow. Great work!!

JuiP committed:
Changes:
- Remove redundant parenthesis
- Fixed line too long
- Update shebang

Saumya-Mishra9129 reviewed (commented):
Reviewed and Tested, works fine. Nice work.

JuiP committed:
Fix : AttributeError
AttributeError : 'IQ' object has no attribute 'SQUARE'
Seen after commit 595df8f

-Reported by Srevin Saju

srevinsaju reviewed (commented):
Good work; works fine; Actually some regressions which were made during my port which I found now... smh

* Mouse cursor is not drawn when cursor moved out of the Sugargame canvas
![image](https://user-images.githubusercontent.com/48695438/85220022-13250d00-b3b1-11ea-9fbc-4b68700c8280.png)

* Occasional error: 
```
Traceback (most recent call last):
  File "/run/media/srevinsaju/ARCH_DATA/git/iq-activity/IQ.py", line 245, in run
    self.do_key(event.key)
  File "/run/media/srevinsaju/ARCH_DATA/git/iq-activity/IQ.py", line 92, in do_key
    if key in self.SQUARE:
AttributeError: 'IQ' object has no attribute 'SQUARE'
```
It happens rarely; One way to reproduce might be to wait for sugar desktop to sleep (the screen goes black) and move the mouse to wake it again. the error is possibly to occur. I will update the comment if I find a method to reproduce it.

srevinsaju reviewed (approved):
Tested as of https://github.com/sugarlabs/iq-activity/pull/20/commits/0b348fa6986bc8524f35df2e8d62424180e18c2a; works perfectly!

JuiP committed:
v23

JuiP commented:
Thanks a lot for reviewing and testing! @srevinsaju @Saumya-Mishra9129 

Done! Ready for release :)
@quozl @chimosky

quozl commented:
Thanks.  Reviewed.  Approved.

chimosky commented:
Tested, thanks.
